### ISSUE TITLE
Add Slack message functionality to send appropriate meeting link

### ISSUE THREAD

**Issue by sarL3y** [hackforla/VRMS#146]
### Overview
A Slack message should be sent to a specific channel including the videoconference link for the upcoming meeting

### Action Items
- [ ] Research Slack API implementation
- [ ] Develop cron job to automatically send Slack message including the videoconference link

**Comment by sarL3y** [comment]
Progress: Successful connection with the Slack API from VRMS.
Next Steps: Pull events per project to figure out appropriate slack channel and notification time to send message
ETA: Currently working, possibly 1 week
Availability: Yes

**Comment by sarL3y** [comment]
Progress: No progress from last week
Next Steps: Pull events per project to figure out appropriate slack channel and notification time to send message
ETA: Will begin tomorrow, 1 week to develop

**Comment by sarL3y** [comment]
On hold for sprint deadline. Will resume after the 21st unless the other work is completed sooner.

**Comment by Jonathanko52** [comment]
Progress: Slack bot messaging test Slack channel
Next Step: Need to know where to find event time/zoom link in database.
ETA:?
Availability: 24/7

**Comment by Jonathanko52** [comment]
Progress: Successfully pulling project slack channel data from database.
Next Step: Setting automatic slack message scheduling.
Blocks: Requires credentials/approval for app to access Hack for LA slack. May be advisable to create test group before this.
ETA: 8/17
Availability:24/7

**Comment by Jonathanko52** [comment]
Progress: Ongoing
Next Step: Setting automatic slack message scheduling.
Blocks: Events being pulled from database aren't showing events for the coming wee, making it difficult to test.
ETA: 8/18
Availability:24/7

**Comment by Jonathanko52** [comment]
Progress: In Progress
Blocks: Database issues
Availability: 24/7
ETA: 8/24

**Comment by Jonathanko52** [comment]
Progress: 
   1.Backend set up to run a check on mondays, making a list of all upcoming events this week.
   2.Function for running events successfully tested.
Blocks:
  1. Need data in the Project collection to identify the slack channels for the upcoming events.
Availability: 
   24/7
ETA: 
   ?

**Comment by alex-anakin** [comment]
@Jonathanko52 please provide your update

**Comment by Jonathanko52** [comment]
Progress: Running as expected on test Slack channel.
Next Step: Testing on Hack for LA Slack Channel
Block: Events collection in database is out of date and not updating properly. No events have been created past 8/26
Availability:24/7

**Comment by Jonathanko52** [comment]
Progress: Reworked to work with service workers.
Next Step: Testing on Hack for LA Slack Channel
Block: Pull request requires reviewing. Requires installation by someone with admin rights to the Hack for LA Slack Channel
ETA: Within this week.

**Comment by alex-anakin** [comment]
@Jonathanko52 
1. The PR is reviewed. 
2. Please ask @ExperimentsInHonesty for help with Hack for LA Slack Channel.

**Comment by Jonathanko52** [comment]
Progress: ExperimentsInHonesty contacted.
Next Step: Create instructions for ExperimentsInHonesty
Block: None. Task in progress.
ETA: 10/5

**Comment by Jonathanko52** [comment]
Progress: Slack Application installed onto Hack for LA Slack Channel.
Next Step: Test App on Hack for LA page.
ETA: 10/6

**Comment by Jonathanko52** [comment]
Progress: New goals added:
-Add comments so other progammers have an easier time figuring out how the new router and worker works.
-Add safeties so testing the app will not duplicate events and post straight to Hack for LA slack if run locally
Next Step: In Progress.
ETA:10/19

**Comment by jbubar** [comment]
@JackHaeg will bring this up with bonnie, to see if this is a priority

**Comment by JackHaeg** [comment]
Closing issue as unplanned, as meeting links being posted in the channel could result in random joiners.
